src/fetchPost.js

```diff
@@ -0,0 +1,365 @@
+            <section class="single-post__note-warning">یادداشت تنها برای شما قابل دیدن است و پس از حذف آگهی، پاک خواهد شد.
+            </section>
+          </section>
+          <a class="single-post__safe-buy iconed-link" href="https://support.divar.ir/b/support-users/fa/kb/articles/article-46">
+                <img class="iconed-link__icon" src="${
+                  require("./icons/svgs/check-circle-o.svg").default
+                }" />
+                <span class="iconed-link__text">راهنمای خرید امن</span>
+          </a>
+          <hr class="single-post__row-separator"/>
+          <a class="single-post__report iconed-link" href="https://support.divar.ir/b/support-users/fa/kb/articles/article-46">
+                <img class="iconed-link__icon" src="${
+                  require("./icons/svgs/info-circle-o.svg").default
+                }" />
+                <span class="iconed-link__text">ثبت تخلف و مشکل آگهی</span>
+          </a>
+          <hr class="single-post__row-separator"/>
+        </section>
+      </article>
+      </main>`.concat(container.innerHTML);
+
+      //carousol handler
+      let res = document.querySelectorAll(".single-post__thumbnail");
+      res.forEach((element) => {
+        element.onclick = function (event) {
+          const image = document.querySelector(".single-post__image");
+          const newImage = event.target.src;
+          image.src = newImage;
+        };
+      });
+
+      //bookmark handler
+      document.querySelector(".single-post__action-bookmark").onclick =
+        function () {
+          let bookmarks = JSON.parse(window.localStorage.getItem("bookmarks"));
+          let icon = document.querySelector(".single-post__bookmark-icon");
+          let bookmarked = false;
+
+          if (bookmarks) {
+            for (let i = 0; i < bookmarks.length; i++) {
+              if (bookmarks[i].url == url) {
+                bookmarked = true;
+              }
+            }
+          }
+
+          if (!icon.classList.contains("bookmarked")) {
+            icon.src = require("./icons/svgs/bookmarked.svg").default;
+            icon.classList.toggle("bookmarked");
+            if (!bookmarked) {
+              if (Array.isArray(bookmarks)) {
+                bookmarks.push({ title: title, url: url });
+              } else {
+                bookmarks = [{ title: title, url: url }];
+              }
+            }
+          } else {
+            icon.src = require("./icons/svgs/bookmark-o.svg").default;
+            icon.classList.toggle("bookmarked");
+            if (Array.isArray(bookmarks) && bookmarked) {
+              bookmarks = bookmarks.filter((item) => item.url !== url);
+            }
+          }
+          window.localStorage.setItem("bookmarks", JSON.stringify(bookmarks));
+        };
+
+      //note handler
+      document.querySelector(".single-post__note").onchange = function (event) {
+        let notes = JSON.parse(window.localStorage.getItem("notes"));
```

Comment:
Alternatively:  
```suggestion
        let notes = JSON.parse(window.localStorage.getItem("notes") || "{}");
```  
JSON.parse throws when given empty string or invalid characters, null is fine though but a try catch would be the safer option
```js
        let parsedNotes;
        let notes = window.localStorage.getItem("notes");
        try {
          parsedNotes = JSON.parse(notes);
        } catch(e) {
          parsedNotes = {};
        }
```  
And after abstraction:  
```js
        let parsedNotes = safeParseJson(window.localStorage.getItem("notes")) || {};
```